chisel/src/main.rs

```diff
@@ -6,65 +6,102 @@ mod driver;
-use libchisel::binaryenopt::*;
-
-use cmd_run::chisel_subcommand_run;
-
-static ERR_NO_SUBCOMMAND: &'static str = "No subcommand provided.";
-
-fn err_exit(msg: &str) -> ! {
-    eprintln!("{}: {}", crate_name!(), msg);
-    process::exit(-1);
-}
-
 pub fn main() {
     let cli_matches = App::new("chisel")
         .version(crate_version!())
         .about(crate_description!())
         .arg(
-            Arg::with_name("VERBOSE")
-                .short("v")
-                .long("verbose")
-                .help("Enables verbose debug logging")
+            Arg::with_name("NO_RECOVER")
+                .short("x")
+                .long("norecover")
+                .help("Exits immediately on all recoverable errors")
                 .global(true),
         )
+        .arg(
+            Arg::with_name("DEBUG_MESSAGES")
+                .short("d")
+                .long("debug")
+                .help("Enables debug messages")
+                .global(true),
+        )
+        .arg(
+            Arg::with_name("MODULES")
+                .short("m")
+                .long("modules")
+                .takes_value(true)
+                .multiple(true)
+                .help("Selects modules to use"),
+        )
+        .arg(Arg::with_name("FILE").help("File to chisel"))
+        .arg(
+            Arg::with_name("OUTPUT_PATH")
+                .short("o")
+                .long("output")
+                .help("Sets the output file in unix mode")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("MODULE_OPTIONS")
+                .short("c")
+                .long("config")
+                .takes_value(true)
+                .multiple(true)
+                .help("Module configuration in unix mode\nConfiguration items come in the form \"module.field=value\"\n\tExample: verifyimports.preset=ewasm"),
+        )
+        .arg(
+            Arg::with_name("OUTPUT_MODE")
+                .long("output-mode")
+                .takes_value(true)
+                .help("Selects the type of output")
+                .possible_values(&["bin", "wat", "hex"])
+                .global(true)
+        )
         .subcommand(
             SubCommand::with_name("run")
-                .about("Runs chisel with the closest configuration file.")
+                .about("Runs chisel in config-driven mode.")
                 .arg(
                     Arg::with_name("CONFIG")
                         .short("c")
                         .long("config")
-                        .help("Sets a custom configuration file")
-                        .value_name("CONF_FILE")
+                        .help("Sets the configuration file in config-driven mode.")
+                        .value_name("PATH")
                         .takes_value(true),
                 ),
         )
+        .after_help("chisel runs in two primary modes: unix-style and config-driven.\n\nunix-style is invoked without a subcommand. \
```

Comment:
Perhaps move this into the oneliner PR too.

Reply:
It's a description for both, I think this should be no big issue.